# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/xhr/XMLHttpRequestWorker.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/xhr/XMLHttpRequestWorker.cpp
--- a/dom/xhr/XMLHttpRequestWorker.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/xhr/XMLHttpRequestWorker.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -290,8 +290,7 @@
 
   MainThreadProxyRunnable(WorkerPrivate* aWorkerPrivate, Proxy* aProxy,
                           const char* aName = "MainThreadProxyRunnable")
-      : MainThreadWorkerSyncRunnable(aWorkerPrivate, aProxy->GetEventTarget(),
-                                     aName),
+      : MainThreadWorkerSyncRunnable(aProxy->GetEventTarget(), aName),
         mProxy(aProxy) {
     MOZ_ASSERT(aProxy);
   }
@@ -777,9 +776,9 @@
       if (mSyncLoopTarget) {
         // We have an unclosed sync loop.  Fix that now.
         RefPtr<MainThreadStopSyncLoopRunnable> runnable =
-            new MainThreadStopSyncLoopRunnable(
-                mWorkerPrivate, std::move(mSyncLoopTarget), NS_ERROR_FAILURE);
-        MOZ_ALWAYS_TRUE(runnable->Dispatch());
+            new MainThreadStopSyncLoopRunnable(std::move(mSyncLoopTarget),
+                                               NS_ERROR_FAILURE);
+        MOZ_ALWAYS_TRUE(runnable->Dispatch(mWorkerPrivate));
       }
 
       mOutstandingSendCount = 0;
@@ -886,7 +885,7 @@
     }
 
     if (runnable) {
-      runnable->Dispatch();
+      runnable->Dispatch(mWorkerPrivate);
     }
   }
 
@@ -924,7 +923,7 @@
 
       RefPtr<ProxyCompleteRunnable> runnable =
           new ProxyCompleteRunnable(mWorkerPrivate, mProxy, mChannelId);
-      if (runnable->Dispatch()) {
+      if (runnable->Dispatch(mWorkerPrivate)) {
         mProxy->mWorkerPrivate = nullptr;
         mProxy->mSyncLoopTarget = nullptr;
         mProxy->mOutstandingSendCount--;